static/js/containers/ProfilePage.js

```diff
@@ -40,9 +39,7 @@ class ProfilePage extends React.Component {
       dispatch(startProfileEdit());
     }
     return dispatch(validateProfile(profile, requiredFields, messages)).then(() => {
-      dispatch(saveProfile(SETTINGS.username, profile)).then(() => {
-        dispatch(clearProfileEdit());
-      });
+      dispatch(saveProfile(SETTINGS.username, profile));
```

Comment:
See the comment above - the problem was not with the `clearProfileEdit` action here, but a syntax error. Since we weren't actually dispatching the `startProfileEdit` action (`startProfileEdit` is a function that returns the action - see the definition in `actions/index.js`) there wasn't an `errors` object to save our validation errors into, and they were being dropped on the floor.